File: TD/TD7/Code/3AB.cpp
/* 1. Procédure -> Afficher le tableau ligne par ligne */
/* 2. Procédure -> Afficher le tableau colonne par colonne */
#include <iostream>

using namespace std;

/* Procédure 1 */

void affParLigne(float tab[5][5])
{
  for (int i = 0; i < 5; ++i) {
    for (int j = 0; j < 5; ++j) {
      std::cout << tab[i][j] << " " << std::flush;
    }
    std::cout << std::endl;
  }
}

/* Procédure 2 */

void affParColonne(float tab[5][5])
{
  for (int i = 0; i < 5; ++i) {
    for (int j = 0; j < 5; ++j) {
      std::cout << tab[j][i] << " " << std::flush;
    }
    std::cout << std::endl;
  }
}

/* Utilisation */

int main()
{
  float tab[5][5] = { 
    {1, 2, 3, 4, 5},
    {1, 2, 3, 4, 5},
    {1, 2, 3, 4, 5},
    {1, 2, 3, 4, 5},
    {1, 2, 3, 4, 5} };
  
  std::cout << "Affichage ligne par ligne" << std::endl;
  affParLigne(tab); // Affichage ligne par ligne
  
  std::cout << "Affichage colonne par colonne" << std::endl;
  affParColonne(tab); // Affichage colonne par colonne
  
  return 0;
}

File: TD/TD7/Code/Python/3AB.py
#!/usr/bin/python
# 1. Afficher le tableau ligne par ligne
# 2. Afficher le tableau colonne par colonne


def aff_par_ligne(tab):
    for i in range(len(tab)):
        for j in range(len(tab[0])):
            print(tab[i][j], '', end='')
        print('\n', end='')


def aff_par_col(tab):
    for i in range(len(tab[0])):
        for j in range(len(tab)):
            print(tab[j][i], '', end='')
        print('\n', end='')


tab = [[1, 2, 3, 4, 5] for i in range(5)]

# Affichage par ligne
print('Affichage ligne par ligne')
aff_par_ligne(tab)

# Affichage par colonne
print('Affichage colonne par colonne')
aff_par_col(tab)
